modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy

```diff
@@ -0,0 +1,664 @@
+import java.nio.file.Paths
+import javax.net.ssl.HostnameVerifier
+import javax.net.ssl.HttpsURLConnection
+import javax.net.ssl.SSLContext
+import javax.net.ssl.TrustManager
+import javax.net.ssl.X509TrustManager
+import nextflow.exception.AbortOperationException
+import nextflow.processor.TaskHandler
+import nextflow.processor.TaskMonitor
+import nextflow.processor.WrMonitor
+import nextflow.processor.TaskRun
+import nextflow.processor.TaskBean
+import nextflow.processor.TaskStatus
+import nextflow.processor.BatchContext
+import nextflow.processor.BatchHandler
+import nextflow.util.Escape
+import groovy.json.JsonSlurper
+import static groovy.json.JsonOutput.toJson
+/**
+ * Executor that schedules jobs using wr as a backend, avoiding storage of
+ * state on disk.
+ *
+ * See https://github.com/VertebrateResequencing/wr
+ *
+ * @author Sendu Bala <sb10@sanger.ac.uk>
+ * Based on TesExecutor by Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ */
+@Slf4j
+@CompileStatic
+class WrExecutor extends Executor {
+
+	static private String token
+	static private String defaultManagerDir
+    static private String endpoint
+	static private String cacertPath
+	static private WrRestApi client
+
+    @Override
+    void register() {
+        if( session.binDir && !session.binDir.empty() ) {
+            session.abort()
+            throw new AbortOperationException("ERROR: wr executor does not allow the use of custom scripts in the `bin` folder")
+        }
+
+        super.register()
+    }
+
+    protected String getDisplayName() {
+        return "$name [$endpoint]"
+    }
+
+	WrRestApi getClient() {
+		if (!client) {
+			defaultManagerDir = Paths.get(System.getProperty('user.home'), ".wr_production")
+			endpoint = getEndPoint()
+			client = new WrRestApi(endpoint, getToken(), getCacertPath())
+		}
+        client
+    }
+
+    protected String getEndPoint() {
+		// default port that wr listens on is 1021 + (uid * 4) + 1
+		// *** note that this will probably only work on linux/mac os, but wr
+		// probably only works fully on those as well...
+		int uid = ["id", "-u"].execute().text.trim() as Integer
+		int port = 1021 + (uid * 4) + 1
+
+        def result = session.getConfigAttribute('executor.endpoint', "https://localhost:$port")
+        log.debug "[wr] endpoint=$result"
+        return result
+    }
+
+	protected String getToken() {
+        String path = session.getConfigAttribute('executor.tokenpath', Paths.get(defaultManagerDir, "client.token"))
+		log.debug "[wr] tokenpath=$path"
+		String result = new File(path).text
+        return result
+    }
+
+	protected String getCacertPath() {
+        String path = session.getConfigAttribute('executor.cacertpath', Paths.get(defaultManagerDir, "ca.pem"))
+		log.debug "[wr] cacertpath=$path"
+        return path
+    }
+
+    /**
+     * @return {@code false} whenever the containerization is managed by the executor itself
+     */
+    boolean isContainerNative() {
+        return false
+    }
```

Comment:
Does `wr` manage the container execution on its own? In that case, it should be `true`.

Reply:
As in doing something if the user wishes to run a docker container? No, wr doesn't yet have native container support. I'm hoping BashWrapperBuilder will put something appropriate in the wrapper script if this boolean returns false.

Reply:
Then it's fine and yes, BashWrapperBuilder should manage the container as usual.